### DISCUSSION TITLE
How to deploy an initially trained model?

### DISCUSSION THREAD

**Discussion by cdeterman** [online-ml/river#1195] [Q&A]
How is it generally recommended to deploy a river model?  Generally, with traditional machine learning models I would save it off as a pickle file, load it when an automated process kicks off and make the predictions.  Is there a recommended saving process that the model is subsequently loaded?  Or is there a particular running process that you normally would see?  I'm curious how people are deploying models such as these.

**Comment by MaxHalford** [comment]
So this is a bit of a deep topic. Basically, there is no one size fits all way to deploy an online model. There are common patterns to learn from, but there are definitely not enough examples out.

This is something I am working on a lot. Sadly I don't have a lot of bandwidth at the moment. A year ago I made [this](https://github.com/MaxHalford/taxi-demo-rp-mz-rv-rd-st) demo, which runs a regression model for predicting taxi trip durations. There's only ~250 lines of code, so it should be a good learning resource. It essentially boils down to loading a model, making predictions, and also learning with it. Honestly it's not rocket science, it's more that there's a lack of tooling out there.

**Comment by cdeterman** [comment]
Have there been any updates in how to manage deployments or should this still be considered open?

**Reply by MaxHalford** [reply]
No nothing new. To be honest I would say this is out of scope of River right now. We've done our best to provide demos/PoCs, but it's up the community to come up with a comprehensive tool.
